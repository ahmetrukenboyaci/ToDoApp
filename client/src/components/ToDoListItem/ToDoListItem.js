/** Dependencies **/
import React, {useCallback} from 'react';
import PropTypes from 'prop-types';

/** Styles **/
import styles from './toDoListItem.module.scss';

/** Images **/
import Uncompleted from '../../assets/images/Uncompleted';
import Completed from '../../assets/images/Completed';
import Delete from '../../assets/images/Delete';

const ToDoListItem = ({data, onDelete, onComplete}) => {
	const onToDoDelete = useCallback(() => {
		onDelete(data._id);
	}, [data?._id, onDelete]);

	const onToDoComplete = useCallback(() => {
		onComplete(data._id, data.done);
	}, [data?._id, data?.done, onComplete]);

	return (
		<div id={'toDoListItemContainer'} className={`${styles.toDoListItemContainer} ${data?.done ? styles.done : ''}`}>
			<div id={'text'} className={styles.text}>
				{data?.description}
			</div>
			<div className={styles.iconContainer}>
				<div onClick={onToDoComplete} className={styles.toggleCompletion}>
					{data?.done ? <Completed height={20}/> : <Uncompleted height={20}/>}
				</div>
				<div onClick={onToDoDelete} className={styles.deleteToDo}>
					<Delete height={20}/>
				</div>
			</div>
		</div>
	);
};

ToDoListItem.propTypes = {
	data: PropTypes.object,
	onDelete: PropTypes.func,
	onComplete: PropTypes.func
};

export default ToDoListItem;