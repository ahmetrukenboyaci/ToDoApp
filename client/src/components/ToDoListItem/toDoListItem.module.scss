.toDoListItemContainer {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 16px;
  margin: 8px;
  border: 1px solid #eee;
  border-radius: 4px;

  &.done {
    background-color: #ddd;
  }

  & .text {

  }

  & .iconContainer {
    display: flex;
    justify-content: center;
    align-items: center;

    & .toggleCompletion {
      display: flex;
      margin-right: 5px;
      cursor: pointer;
    }

    & .deleteToDo {
      display: flex;
      margin-left: 5px;
      cursor: pointer;
    }
  }
}