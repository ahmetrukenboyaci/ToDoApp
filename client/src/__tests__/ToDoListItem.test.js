import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import {configure, shallow} from 'enzyme';
import ToDoListItem from '../components/ToDoListItem/ToDoListItem';

configure({adapter: new Adapter()});

describe('ToDoListItem', () => {
	it('should render todolistitem component', () => {
		// eslint-disable-next-line no-unused-vars
		const wrapper = shallow(<ToDoListItem/>);
	});

	it('should render initial layout', () => {

		const component = shallow(<ToDoListItem/>);

		expect(component.getElements()).toMatchSnapshot();
	});

	it('should render text', () => {
		const data = {_id: '2', description: 'b', done: false};

		const component = shallow(<ToDoListItem data={data}/>);

		const text = component.find('#text');

		expect(text.text()).toEqual(data.description);
	});
});