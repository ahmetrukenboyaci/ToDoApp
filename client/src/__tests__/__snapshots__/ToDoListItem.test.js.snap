// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToDoListItem should render initial layout 1`] = `
Array [
  <div
    className="toDoListItemContainer "
    id="toDoListItemContainer"
  >
    <div
      className="text"
      id="text"
    />
    <div
      className="iconContainer"
    >
      <div
        className="toggleCompletion"
        onClick={[Function]}
      >
        <Uncompleted
          height={20}
        />
      </div>
      <div
        className="deleteToDo"
        onClick={[Function]}
      >
        <Delete
          height={20}
        />
      </div>
    </div>
  </div>,
]
`;
